### ISSUE TITLE
TypeError: locale.split is not a function

### ISSUE THREAD

**Issue by thomas-cruz** [fnando/i18n-js#543]
Using react native 0.57. 
  `componentDidMount() {`
    `I18n.defaultLocale = "en";`
   ` I18n.locale = this.getStoredLanguage();`
    `I18n.currentLocale();`
   `....`
  `} `
All I'm trying to do is fetch whatever locale was set from wherever the persistent data is being stored and set that as the current locale.

Here is the full error log, I don't know if this is helpful
TypeError: TypeError: TypeError: TypeError: locale.split is not a function

This error is located at:
    in Main (created by Connect(Main))
    in Connect(Main) (at navigationStore.js:448)
    in Wrapped (at SceneView.js:9)
    in SceneView (at StackViewLayout.js:478)
    in RCTView (at View.js:44)
    in RCTView (at View.js:44)
    in RCTView (at View.js:44)
    in AnimatedComponent (at screens.native.js:59)
    in Screen (at StackViewCard.js:42)
    in Card (at createPointerEventsContainer.js:26)
    in Container (at StackViewLayout.js:507)
    in RCTView (at View.js:44)
    in ScreenContainer (at StackViewLayout.js:401)
    in RCTView (at View.js:44)
    in StackViewLayout (at withOrientation.js:30)
    in withOrientation (at StackView.js:49)
    in RCTView (at View.js:44)
    in Transitioner (at StackView.js:19)
    in StackView (at createNavigator.js:57)
    in Navigator (at createKeyboardAwareNavigator.js:11)
    in KeyboardAwareNavigator (at createNavigationContainer.js:376)
    in NavigationContainer (at SceneView.js:9)
    in SceneView (at createTabNavigator.js:39)
    in RCTView (at View.js:44)
    in RCTView (at View.js:44)
    in ResourceSavingScene (at createBottomTabNavigator.js:108)
    in RCTView (at View.js:44)
    in ScreenContainer (at createBottomTabNavigator.js:98)
    in RCTView (at View.js:44)
    in TabNavigationView (at createTabNavigator.js:178)
    in NavigationView (at createNavigator.js:57)
    in Navigator (at createNavigationContainer.js:376)
    in NavigationContainer (at SceneView.js:9)
    in SceneView (at StackViewLayout.js:478)
    in RCTView (at View.js:44)
    in RCTView (at View.js:44)
    in RCTView (at View.js:44)
    in AnimatedComponent (at screens.native.js:59)
    in Screen (at StackViewCard.js:42)
    in Card (at createPointerEventsContainer.js:26)
    in Container (at StackViewLayout.js:507)
    in RCTView (at View.js:44)
    in ScreenContainer (at StackViewLayout.js:401)
    in RCTView (at View.js:44)
    in StackViewLayout (at withOrientation.js:30)
    in withOrientation (at StackView.js:49)
    in RCTView (at View.js:44)
    in Transitioner (at StackView.js:19)
    in StackView (at createNavigator.js:57)
    in Navigator (at createKeyboardAwareNavigator.js:11)
    in KeyboardAwareNavigator (at createNavigationContainer.js:376)
    in NavigationContainer (at Router.js:97)
    in App (at Router.js:122)
    in Router (at AppRouter.js:242)
    in AppRouter (at App.js:47)
    in PersistGate (at App.js:46)
    in Provider (at App.js:45)
    in App (at renderApplication.js:34)
    in RCTView (at View.js:44)
    in RCTView (at View.js:44)
    in AppContainer (at renderApplication.js:33)

This error is located at:
    in NavigationContainer (at SceneView.js:9)
    in SceneView (at createTabNavigator.js:39)
    in RCTView (at View.js:44)
    in RCTView (at View.js:44)
    in ResourceSavingScene (at createBottomTabNavigator.js:108)
    in RCTView (at View.js:44)
    in ScreenContainer (at createBottomTabNavigator.js:98)
    in RCTView (at View.js:44)
    in TabNavigationView (at createTabNavigator.js:178)
    in NavigationView (at createNavigator.js:57)
    in Navigator (at createNavigationContainer.js:376)
    in NavigationContainer (at SceneView.js:9)
    in SceneView (at StackViewLayout.js:478)
    in RCTView (at View.js:44)
    in RCTView (at View.js:44)
    in RCTView (at View.js:44)
    in AnimatedComponent (at screens.native.js:59)
    in Screen (at StackViewCard.js:42)
    in Card (at createPointerEventsContainer.js:26)
    in Container (at StackViewLayout.js:507)
    in RCTView (at View.js:44)
    in ScreenContainer (at StackViewLayout.js:401)
    in RCTView (at View.js:44)
    in StackViewLayout (at withOrientation.js:30)
    in withOrientation (at StackView.js:49)
    in RCTView (at View.js:44)
    in Transitioner (at StackView.js:19)
    in StackView (at createNavigator.js:57)
    in Navigator (at createKeyboardAwareNavigator.js:11)
    in KeyboardAwareNavigator (at createNavigationContainer.js:376)
    in NavigationContainer (at Router.js:97)
    in App (at Router.js:122)
    in Router (at AppRouter.js:242)
    in AppRouter (at App.js:47)
    in PersistGate (at App.js:46)
    in Provider (at App.js:45)
    in App (at renderApplication.js:34)
    in RCTView (at View.js:44)
    in RCTView (at View.js:44)
    in AppContainer (at renderApplication.js:33)

This error is located at:
    in NavigationContainer (at SceneView.js:9)
    in SceneView (at StackViewLayout.js:478)
    in RCTView (at View.js:44)
    in RCTView (at View.js:44)
    in RCTView (at View.js:44)
    in AnimatedComponent (at screens.native.js:59)
    in Screen (at StackViewCard.js:42)
    in Card (at createPointerEventsContainer.js:26)
    in Container (at StackViewLayout.js:507)
    in RCTView (at View.js:44)
    in ScreenContainer (at StackViewLayout.js:401)
    in RCTView (at View.js:44)
    in StackViewLayout (at withOrientation.js:30)
    in withOrientation (at StackView.js:49)
    in RCTView (at View.js:44)
    in Transitioner (at StackView.js:19)
    in StackView (at createNavigator.js:57)
    in Navigator (at createKeyboardAwareNavigator.js:11)
    in KeyboardAwareNavigator (at createNavigationContainer.js:376)
    in NavigationContainer (at Router.js:97)
    in App (at Router.js:122)
    in Router (at AppRouter.js:242)
    in AppRouter (at App.js:47)
    in PersistGate (at App.js:46)
    in Provider (at App.js:45)
    in App (at renderApplication.js:34)
    in RCTView (at View.js:44)
    in RCTView (at View.js:44)
    in AppContainer (at renderApplication.js:33)

This error is located at:
    in NavigationContainer (at Router.js:97)
    in App (at Router.js:122)
    in Router (at AppRouter.js:242)
    in AppRouter (at App.js:47)
    in PersistGate (at App.js:46)
    in Provider (at App.js:45)
    in App (at renderApplication.js:34)
    in RCTView (at View.js:44)
    in RCTView (at View.js:44)
    in AppContainer (at renderApplication.js:33)
<unknown>
    i18n.js:312:31
I18n.locales.default
    i18n.js:311:12
Object.I18n.locales.get
    i18n.js:239:15
Object.I18n.lookup
    i18n.js:394:31
Object.<anonymous>
    i18n.js:589:29
Object.I18n.translate
    i18n.js:586:25
Main.render
    Main.js:690:13
Main.proxiedMethod
    createPrototypeProxy.js:44:29

And here's an image of the error on an Android Emulator (version 9.0)
![Screenshot_1566188602](https://user-images.githubusercontent.com/50308862/63239494-065df700-c27e-11e9-8b07-33f1ea927cb8.png)

**Comment by thomas-cruz** [comment]
Work around
in node_modules/i18n-js/app/assets/javascripts/i18n.js, line 312
` locales.forEach(function(locale) {`
`      var localeParts = String(locale).split("-"); //forced type String on locale` 
`      var firstFallback = null;`
`      var secondFallback = null;`
`      if (localeParts.length === 3) {`
` ... `
only problem now is the locale not changing after being set but that's probably a different issue

**Comment by PikachuEXE** [comment]
`locale.split is not a function`
means the "type/class" of `locale` is updated to a non `string` value
I can't help you debug since I don't know what can be returned by `this.getStoredLanguage()`

**Comment by InstantProgramming** [comment]
I just had to deal with this issue.
The issue is coming like you supposed from this.getStoredLanguage(), this is probably returning a Promise and not a proper value. 
I managed to fix it by creating a State with a null value at construction state={locale: null} , then in componentDidMount, i called this.getStoredLanguage().then(value=>this.setState({locale: value})).

If it's the same issue, it's not comming from i18n-js.
